devito/interfaces.py

```diff
@@ -127,7 +132,10 @@ def set_initializer(self, lambda_initializer):
         self.initializer = lambda_initializer
 
     def _allocate_memory(self):
-        self._data = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        if self.memmap:
+            self._data = np.memmap(filename=(self.f), dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C')
```

Comment:
Why are we talking about alignment for mmap. Isn't the start address always
on a page boundary?


On Fri, 15 Jul 2016 at 11:38 pp1336 <notifications@github.com> wrote:

> In devito/interfaces.py
> <https://github.com/opesci/devito/pull/36#discussion_r70953718>:
>
> > @@ -127,7 +132,10 @@ def set_initializer(self, lambda_initializer):
> >          self.initializer = lambda_initializer
> >
> >      def _allocate_memory(self):
> > -        self._data = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
> > +        if self.memmap:
> > +            self._data = np.memmap(filename=(self.f), dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C')
>
> The () is just to make it easier to read I can remove it if you like.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/opesci/devito/pull/36/files/1025b73ec6154e1a457d1e57c9e4bdf22af36f73#r70953718>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFJRA3viG0TJfppljO_uA35YPBAlC44Gks5qV2MqgaJpZM4JJiBx>
> .
>